File: src/main/java/org/afterlike/lucid/platform/command/LucidCommand.java
package org.afterlike.lucid.platform.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;

public class LucidCommand extends CommandBase {
	@Override
	public String getCommandName() {
		return "lucid";
	}

	@Override
	public String getCommandUsage(ICommandSender sender) {
		return "/lucid";
	}

	@Override
	public int getRequiredPermissionLevel() {
		return 0;
	}

	@Override
	public void processCommand(ICommandSender sender, String[] args) throws CommandException {
		// todo: add gui
	}
}

File: src/main/kotlin/org/afterlike/lucid/core/command/LucidCommand.kt
package org.afterlike.lucid.core.command

import net.minecraft.client.Minecraft
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import org.afterlike.lucid.core.handler.DelayedTaskHandler
import org.afterlike.lucid.gui.screen.LucidGuiScreen

class LucidCommand : CommandBase() {
    override fun getCommandName(): String = "lucid"

    override fun getCommandUsage(sender: ICommandSender): String = "/lucid"

    override fun getRequiredPermissionLevel(): Int = 0

    override fun processCommand(sender: ICommandSender, args: Array<String>) {
        DelayedTaskHandler.schedule(1) {
            Minecraft.getMinecraft().displayGuiScreen(LucidGuiScreen())
        }
    }
}